components/brave_shields/browser/ad_block_default_source_provider.h

```diff
@@ -0,0 +1,65 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_COMPONENTS_BRAVE_SHIELDS_BROWSER_AD_BLOCK_DEFAULT_SOURCE_PROVIDER_H_
+#define BRAVE_COMPONENTS_BRAVE_SHIELDS_BROWSER_AD_BLOCK_DEFAULT_SOURCE_PROVIDER_H_
+
+#include <string>
+
+#include "base/callback.h"
+#include "base/observer_list.h"
+#include "brave/components/brave_component_updater/browser/dat_file_util.h"
+#include "brave/components/brave_shields/browser/ad_block_resource_provider.h"
+#include "brave/components/brave_shields/browser/ad_block_source_provider.h"
+#include "third_party/abseil-cpp/absl/types/optional.h"
+
+using brave_component_updater::DATFileDataBuffer;
+
+namespace component_updater {
+class ComponentUpdateService;
+}  // namespace component_updater
+
+namespace base {
+class FilePath;
+}
+
+class AdBlockServiceTest;
+
+namespace brave_shields {
+
+class AdBlockDefaultSourceProvider : public SourceProvider,
+                                     public ResourceProvider {
+ public:
+  AdBlockDefaultSourceProvider(
+      component_updater::ComponentUpdateService* cus,
+      base::RepeatingCallback<void(const std::string& regional_catalog)>
+          regional_catalog_available_cb);
+  ~AdBlockDefaultSourceProvider() override;
+
+  void Load(base::OnceCallback<
```

Comment:
don't overload method names that do different things. One of these loads a dat file and the other loads resources

Reply:
they are technically overridenf from `AdBlockSourceProvider` and `AdBlockResourceProvider` respectively; they make sense in context there but maybe not when overloaded side-by-side. I can rename to `LoadDATBuffer` and `LoadResources` respectively